### ISSUE TITLE
\\wsl$ filesystem does not support file locking

### ISSUE THREAD

**Issue by patriklantto** [microsoft/WSL#4689]
Trying to use SVN from Windows on the \\wsl$ drive, svn reports the following:

svn: E200033: Another process is blocking the working copy database, or the underlying filesystem does not support file locking; if the working copy is on a network filesystem, make sure file locking has been enabled on the file server
svn: E200033: sqlite[S5]: database is locked, executing statement 'PRAGMA case_sensitive_like=1;PRAGMA synchronous=OFF;PRAGMA recursive_triggers=ON;PRAGMA foreign_keys=OFF;PRAGMA locking_mode = NORMAL;PRAGMA journal_mode = TRUNCATE;'

I assume the problem is that the "underlying filesystem does not support file locking" - is this assumption correct, and is there a way to address this?

**Comment by patriklantto** [comment]
Information according to the ISSUE_TEMPLATE:

- Microsoft Windows [Version 10.0.19013.1122]
- Mount \wsl$\Ubuntu as U:
- Try to checkout a SVN repository into a directory under U: using "svn co <repository>"
- svn will fail with the following message:

svn: E200033: Another process is blocking the working copy database, or the underlying filesystem does not support file locking; if the working copy is on a network filesystem, make sure file locking has been enabled on the file server

- svn version is 1.13.0 (r1867053)

**Comment by MartijnJonkers** [comment]
Did you find a solution? I am facing the same problem?

**Comment by patriklantto** [comment]
> Did you find a solution? I am facing the same problem.

Unfortunately not :-(

The reason I would like to use SVN from the Windows environment and not WSL is that we have some Windows tools for SVN operations - and for now I am using rsync to move files between the two filesystems when I need to use these tools.

**Comment by luisfdgon** [comment]
Hi guys,

Did you find any solution meanwhile? This is really a mess not being able to use SVN.

**Comment by GbuHti** [comment]
Hi, you can install a SVN client on you WSL(sudo apt install subversion), and use this SVN to checkout...

**Comment by oxidaan** [comment]
Yes you can use a SVN client in WSL2, but it is a pitty not to be able to use TortoiseSVN which is much more powerfull imho. Issue seems related to https://github.com/sqlitebrowser/sqlitebrowser/issues/2142 and kind of the opposite of wsl1 issue: https://github.com/microsoft/WSL/issues/2395

**Comment by makerlabs** [comment]
Alternatively u can install additional file synchronization software line im using https://syncthing.net/ its easy to set up and use on WSL and Windows Thanks to this setup I can use with no problem any available SVN software on my windows machine.

**Comment by hsali** [comment]
when this going to fixed. any idea?

**Comment by hsali** [comment]
I tried using pycharm and sqlite db. jetbrain depended to fix their side. 

https://youtrack.jetbrains.com/issue/DBE-11014/Datagrip-Support-for-WSL-paths-SQLite#focus=Comments-27-6073328.0-0

**Comment by rinormaloku** [comment]
We need this

**Comment by vbirds** [comment]
We need this

**Comment by olimexsmart** [comment]
Same problem here with SQLite

**Comment by johan-laneau** [comment]
The work-around I use is to add a Samba share from the root of the WSL2 filesystem. I use following commands in Ubuntu
`# Install samba server
export DEBIAN_FRONTEND=noninteractive && \
sudo apt update && \
sudo apt install --no-install-recommends --yes samba && \

# Configure the samba server
sudo bash -c "cat >> /etc/samba/smb.conf" << EOT
[wslroot]
    comment = Root of the wsl machine
    path = /
    read only = no
    browsable = yes
EOT

# Set the password for the samba-user
echo "mypass
mypass" | sudo smbpasswd -a -s ${USER} && \

# (Re)-start the samba server with the updated configuration
sudo service smbd restart && \

echo "Samba share for wslroot installed successfully!"
`
Then you can map this share to a network drive.
On the drive, you can use SVN or TortoiseSVN as you want. Just note that line endings might be different depending upon your setup.

The fact that the samba share allows to work-around means that the real root cause of this issue is in the protocol that Microsoft uses for the WSL2 file sharing. Not sure if this is caused by the choice for Plan 9 (https://nelsonslog.wordpress.com/2019/06/01/wsl-access-to-linux-files-via-plan-9/).

**Comment by jhimes144** [comment]
This is still an issue for developers testing dockerized linux apps on Windows that use SqlLite databases, because this requires a docker volume on WSL, which exhibits this issue, making it impossible to work with SQLLite.

**Comment by dionisvl** [comment]
up

**Comment by davidedelerma** [comment]
up

**Comment by PylotLight** [comment]
Also having this issue: https://github.com/pocketbase/pocketbase/discussions/4367#discussioncomment-8516929

**Comment by g10-fred** [comment]
Still an issue in PHP Storm

**Comment by tddouglas** [comment]
Still encountering on PyCharm

**Comment by fliespl** [comment]
Still valid

**Comment by strawberrymelonpanda** [comment]
Ran into this just today with SQLite browsers. 😞

**Comment by dionisvl** [comment]
up

**Comment by rustam-python** [comment]
Up

**Comment by ndn1991** [comment]
Need this

**Comment by traktorlixach** [comment]
up

**Comment by musagenius345** [comment]
up

**Comment by Melochaotic** [comment]
Adding one more plea in 2025. Hopefully the year for a fix 🤞

**Comment by zkw13** [comment]
need this

**Comment by mariagdcGroup** [comment]
It may be helpful to someone. I have fixed the issue with PHPStorm connection by adding '`file`' before the path and    
'`?nolock`' param after. Also, there are backslashes in the path. 
 
Example: 

` jdbc:sqlite:file:\\wsl$\Ubuntu\path\to\database\file\database.sqlite?nolock=1`

**Comment by ignatov-dmitry** [comment]
> It may be helpful to someone. I have fixed the issue with PHPStorm connection by adding '`file`' before the path and '`?nolock`' param after. Also, there are backslashes in the path.
> 
> Example:
> 
> ` jdbc:sqlite:file:\\wsl$\Ubuntu\path\to\database\file\database.sqlite?nolock=1`

thanks a lot) You saved me a lot of time)
